ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.websocket;
+
+import android.support.annotation.Nullable;
+
+import java.io.IOException;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+import com.squareup.okhttp.ws.WebSocket;
+import com.squareup.okhttp.ws.WebSocketCall;
+import com.squareup.okhttp.ws.WebSocketListener;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import okio.Buffer;
+import okio.BufferedSource;
+
+public class WebSocketModule extends ReactContextBaseJavaModule {
+  private Map<Integer, WebSocket> mWebSocketConnections = new HashMap<>();
+
+  private ReactContext reactContext;
+
+  public WebSocketModule(ReactApplicationContext ctx) {
+    super(ctx);
+
+    reactContext = ctx;
+  }
+
+  private void sendEvent(String eventName, @Nullable WritableMap params) {
+    reactContext
+            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+            .emit(eventName, params);
+  }
+
+  @Override
+  public String getName() {
+    return "WebSocketManager";
+  }
+
+  @ReactMethod
+  public void connect(final String url, final int id) {
+    OkHttpClient client = new OkHttpClient();
+
+    Request request = new Request.Builder()
+        .tag(id)
+        .url(url)
+        .build();
+
+    WebSocketCall.create(client, request).enqueue(new WebSocketListener() {
+      @Override
+      public void onOpen(final WebSocket webSocket, final Response response) {
+        mWebSocketConnections.put(id, webSocket);
+
+        WritableMap params = Arguments.createMap();
+
+        params.putInt("id", id);
+
+        sendEvent("websocketOpen", params);
+      }
+
+      @Override
+      public void onMessage(
+          final BufferedSource bufferedSource,
+          final WebSocket.PayloadType payloadType
+      ) {
+        String message = "";
+
+        try {
+          message = bufferedSource.readUtf8();
+        } catch (IOException e) {
+          e.printStackTrace();
```

Comment:
This seems like a pretty serious error to ignore. In general you have a few ways to do error handling:
- If it's an actionable error for the developer, pass the error to JS. In this case that means invoking `onFailure`.
- If it's an error the developer has no good way of handling, don't catch the exception. It will be caught at the [framework level](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/bridge/CatalystInstance.java#L73) and cause a redbox in dev mode and a hard crash in production.
- If it's an error that you want to swallow (and e.g. convert to some default return value) it should always be logged with `FLog.e` (see [RequestBodyUtil](https://github.com/facebook/react-native/blob/42eb5464fd8a65ed84b799de5d4dc225349449be/ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java#L59)), that way it will have the correct tag in logcat, as opposed to when it's logged to stdout using `e.printStackTrace();`.

In this case it's a network error so I'd go with the first option.